### ISSUE TITLE
Intermittent test failure with `make test`

### ISSUE THREAD

**Issue by arichardet** [asteris-llc/converge#476]
Running `make test` results with this failure sometimes.

```
=== testing blackbox/test_apply_sourcefile_remote.sh ===
2016-11-04T18:42:38Z |WARN| no token set, server is unauthenticated. This should *only* be used for development.
2016-11-04T18:42:38Z |WARN| no SSL config in use, server will accept HTTP connections
2016-11-04T18:42:38Z |INFO| serving	addr=:4774 component=rpc
2016-11-04T18:42:38Z |INFO| serving	addr=:2694 protocol=HTTP service=API
/tmp/converge.apply.TH2K8UgfJz /vagrant/godev/src/github.com/asteris-llc/converge
2016-11-04T18:42:39Z |INFO| serving	addr=:47740 component=rpc
2016-11-04T18:42:39Z |WARN| skipping module verification	component=client
2016-11-04T18:42:39Z |INFO| got status	component=client file=http://localhost:2694/api/v1/resources/modules/sourceFile.hcl id=root/param.message run=STARTED stage=APPLY
2016-11-04T18:42:39Z |INFO| got status	component=client file=http://localhost:2694/api/v1/resources/modules/sourceFile.hcl id=root/module.basic/param.filename run=STARTED stage=APPLY
2016-11-04T18:42:39Z |INFO| got status	component=client file=http://localhost:2694/api/v1/resources/modules/sourceFile.hcl id=root/module.basic/param.message run=STARTED stage=APPLY
2016-11-04T18:42:39Z |INFO| got status	component=client file=http://localhost:2694/api/v1/resources/modules/sourceFile.hcl id=root/module.basic/task.render run=STARTED stage=APPLY
2016-11-04T18:42:39Z |INFO| got status	component=client file=http://localhost:2694/api/v1/resources/modules/sourceFile.hcl id=root/module.basic run=STARTED stage=APPLY
2016-11-04T18:42:39Z |INFO| got status	component=client file=http://localhost:2694/api/v1/resources/modules/sourceFile.hcl id=root run=STARTED stage=APPLY

root/module.basic/task.render:
 Error: execution timed out
 Messages:
 Has Changes: no
 Changes: No changes

Summary: 3 errors, 0 changes

 * root/module.basic/task.render: execution timed out
 * root/module.basic: error in dependency "root/module.basic/task.render"
 * root: error in dependency "root/module.basic"
make: *** [blackbox/test_apply_sourcefile_remote.sh] Error 1
2016-11-04T18:42:39Z |INFO| gracefully shutting down (interrupt again to halt)
2016-11-04T18:42:39Z |INFO| halted	addr=:2694 service=API
```

**Comment by arichardet** [comment]
It looks like this is a known bug referenced in resource/shell/command_executor.go:
```
// Currently when a script executes beyond it's alloted time a timeout will
// occur and nil is returned by timeoutExec.  The goroutine running the script
// will continue to drain stdout and stderr from the process sockets and they
// will be GCed when the script finally finishes.  This means that there is no
// mechanism for getting the output of a script when it has timed out.  A
// proposed solution to this would be to implement a ReadUntilWouldBlock-type
// function that would allow us to read into a buffer from a ReadCloser until
// the read operation would block, then return the contents of the buffer (along
// with some marker if we recevied an error or EOF).  Then exec function would
// then take in pointers to buffers for stdout and stderr and populate them
// directly, so that if the script execution timed out we would still have a
// reference to those buffers.
```

**Comment by arichardet** [comment]
Fixed by #483
